pyo3-macros-backend/src/pyclass.rs

```diff
@@ -652,6 +660,13 @@ impl EnumVariantPyO3Options {
                     );
                     options.name = Some(name);
                 }
+                EnumVariantPyO3Option::Constructor(constructor) => {
+                    ensure_spanned!(
+                        options.constructor.is_none(),
+                        constructor.span() => "`constructor` may only be specified once"
+                    );
+                    options.constructor = Some(constructor);
+                }
```

Comment:
Possibly worth seeing `PyClassPyO3Options::set_option`, which has a `set_option!` macro helper to reduce repetition. If you like that pattern we could reuse it here.

Reply:
That looks nice, I'll adopt that here to make future extending easier 👍